go/arrow/array/table.go

```diff
@@ -198,6 +198,26 @@ func NewTableFromRecords(schema *arrow.Schema, recs []arrow.Record) *simpleTable
 }
 
 func (tbl *simpleTable) Schema() *arrow.Schema      { return tbl.schema }
+
+func (tbl *simpleTable) AddColumn(i int, field arrow.Field, column arrow.Column) (*simpleTable, error) {
```

Comment:
Would `InsertColumn` be a better name? You also need to add this to the `Table` interface.

When you do, please add a documentation comment to clarify the semantics that this creates a new table that needs to be released instead of modifying the existing table.